service.subtitles.subtitool/service.py

```diff
@@ -0,0 +1,216 @@
+    return d
+
+
+def f(z):
+    idx = [0xe, 0x3, 0x6, 0x8, 0x2]
+    mul = [2, 2, 5, 4, 3]
+    add = [0, 0xd, 0x10, 0xb, 0x5]
+
+    b = []
+    for i in xrange(len(idx)):
+        a = add[i]
+        m = mul[i]
+        i = idx[i]
+
+        t = a + int(z[i], 16)
+        v = int(z[t:t + 2], 16)
+        b.append(("%x" % (v * m))[-1])
+
+    return ''.join(b)
+
+
+def Search(item,langs):
+    d = timeout(set_filehash, args=(item["file_original_path"], item["rar"]), timeout_duration=15)
+    md5hash = d.hexdigest()
+    t = f(md5hash)
+    filename = '.'.join(os.path.basename(item["file_original_path"]).split(".")[:-1])
+    helper = SubTiToolHelper(filename, md5hash)
+    results = helper.search(item, t, langs)
+
+    results = results.getElementsByTagName('Subtitle')
+    for node in results:
+
+        sTitle = node.getElementsByTagName('TITLE')[0].firstChild.data
+        sLang = node.getElementsByTagName('LANGUAGE')[0].firstChild.data
+        sTitle = node.getElementsByTagName('TITLE')[0].firstChild.data
+        sDlLink = node.getElementsByTagName('DOWNLOADLINK')[0].firstChild.data
+        sRate = node.getElementsByTagName('RATE')[0].firstChild.data
+        if sLang == "Farsi/Persian": sLang = "Persian"
+
+        listitem = xbmcgui.ListItem(label=xbmc.convertLanguage(sLang, xbmc.ENGLISH_NAME),
+                                    # language name for the found subtitle
+                                    label2=sTitle,  # file name for the found subtitle
+                                    iconImage=sRate,  # rating for the subtitle, string 0-5
+                                    thumbnailImage=xbmc.convertLanguage(sLang, xbmc.ISO_639_1)
+                                    # language flag, ISO_639_1 language + gif extention, e.g - "en.gif"
+                                    )
+        listitem.setProperty("sync", '{0}'.format("true").lower())  # set to "true" if subtitle is matched by hash,
+        # indicates that sub is 100 Comaptible
+        listitem.setProperty("hearing_imp",
+                             '{0}'.format("false").lower())  # set to "true" if subtitle is for hearing impared
+
+        ## below arguments are optional, it can be used to pass any info needed in download function
+        ## anything after "action=download&" will be sent to addon once user clicks listed subtitle to download
+        url = "plugin://%s/?action=download&l=%s&f=%s&filename=%s&dllink=%s" % (
+            __scriptid__, sLang, md5hash, filename, sDlLink)
+        ## add it to list, this can be done as many times as needed for all subtitles found
+        xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=url, listitem=listitem, isFolder=False)
+
+
+def Download(language, hash, filename, dlLink):
+    subtitle_list = []
+    ## Cleanup temp dir, we recomend you download/unzip your subs in temp folder and
+    ## pass that to XBMC to copy and activate
+    if xbmcvfs.exists(__temp__):
+        shutil.rmtree(__temp__)
```

Comment:
did you try xbmcvfs.rmdir()?
https://romanvm.github.io/Kodistubs/_autosummary/xbmcvfs.html#xbmcvfs.rmdir